DatadogSessionReplay/Sources/Recorder/ViewTreeSnapshotProducer/ViewTreeSnapshot/NodeRecorders/SwiftUI/SwiftUIWireframesBuilder.swift

```diff
@@ -0,0 +1,162 @@
+            switch item.value {
+            case let .effect(effect, list):
+                return wireframes + effectWireframe(item: item, effect: effect, list: list, context: context)
+            case .content(let content):
+                return wireframes + contentWireframe(item: item, content: content, context: context)
+            case .unknown:
+                return wireframes
+            }
+        }
+    }
+
+    private func contentWireframe(item: DisplayList.Item, content: DisplayList.Content, context: Context) -> SRWireframe? {
+        let viewInfo = renderer.viewCache.map[.init(id: .init(identity: item.identity))]
+
+        switch content.value {
+        case let .shape(_, paint, _):
+            return paint.paint.map { paint in
+                context.builder.createShapeWireframe(
+                    id: Int64(content.seed.value),
+                    frame: context.convert(frame: item.frame),
+                    clip: context.clip,
+                    backgroundColor: CGColor(
+                        red: CGFloat(paint.linearRed),
+                        green: CGFloat(paint.linearGreen),
+                        blue: CGFloat(paint.linearBlue),
+                        alpha: CGFloat(paint.opacity)
+                    ),
+                    cornerRadius: viewInfo?.cornerRadius,
+                    opacity: CGFloat(paint.opacity)
+                )
+            }
+        case let .text(view, _):
+            let storage = view.text.storage
+            let style = storage.attribute(.paragraphStyle, at: 0, effectiveRange: nil) as? NSParagraphStyle
+            let foregroundColor = storage.attribute(.foregroundColor, at: 0, effectiveRange: nil) as? UIColor
+            let font = storage.attribute(.font, at: 0, effectiveRange: nil) as? UIFont
+            return context.builder.createTextWireframe(
+                id: Int64(content.seed.value),
+                frame: context.convert(frame: item.frame),
+                clip: context.clip,
+                text: textObfuscator.mask(text: storage.string),
+                textAlignment: style.map { .init(systemTextAlignment: $0.alignment) },
+                textColor: foregroundColor?.cgColor,
+                font: font,
+                fontScalingEnabled: fontScalingEnabled
+            )
+        case .color:
+            return context.builder.createShapeWireframe(
+                id: Int64(content.seed.value),
+                frame: context.convert(frame: item.frame),
+                clip: context.clip,
+                borderColor: viewInfo?.borderColor,
+                borderWidth: viewInfo?.borderWidth,
+                backgroundColor: viewInfo?.backgroundColor,
+                cornerRadius: viewInfo?.cornerRadius,
+                opacity: viewInfo?.alpha
+            )
+        case .platformView:
+            return nil // Should be recorder by UIKit recorder
+        case .unknown:
+            return nil // Need a placeholder
```

Comment:
Should we return an "unsupported" placeholder wireframe then?

Reply:
Yes, lets add this in further PR